src/Transport/Topology/MetaModel/NamespacesDefinition.cs

```diff
@@ -0,0 +1,69 @@
+﻿namespace NServiceBus.AzureServiceBus
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    public class NamespacesDefinition : IEnumerable<NamespaceDefinition>
+    {
+        private static readonly string DefaultName = "default";
+        private static readonly string Prefix = "namespace-";
+
+        private readonly List<NamespaceDefinition> _inner;
+
+        public NamespacesDefinition()
+        {
+            _inner = new List<NamespaceDefinition>();
+        }
+
+        public int Count => _inner.Count;
+
+        public void Add(string name, string connectionString)
+        {
+            var definition = new NamespaceDefinition(name, connectionString);
+            if (_inner.Contains(definition)) return;
+
+            var defaultDefinition = _inner.SingleOrDefault(x => x.Name == DefaultName && x.ConnectionString == connectionString);
+            _inner.Remove(defaultDefinition);
+            _inner.Add(definition);
```

Comment:
Through config API, users can setup namespaces in two different ways: adding namespace explicitly or using ConnectionString core extension method.
In the develop branch version, throughout transport set up, we add "default" connection string to namespaces if it doesn't exist. I put this logic inside NamespacesDefinition class: adding a new namespace definition (name + connectionString), if an identical connection string marked as default already exists, I replace it with a more specific namespace definition